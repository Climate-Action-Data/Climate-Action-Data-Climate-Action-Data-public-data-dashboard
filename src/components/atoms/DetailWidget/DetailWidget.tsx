import { BoxProps, Box, Text } from '@chakra-ui/react'

interface DetailWidgetProps extends BoxProps {
  title: string
  asBox?: boolean
}

const extractDesignProps = (props: DetailWidgetProps): Partial<BoxProps> => {
  const designProps: Partial<DetailWidgetProps> = { ...props }
  if (designProps.title !== undefined) {
    delete designProps.title
  }
  if (designProps.asBox !== undefined) {
    delete designProps.asBox
  }
  return designProps as BoxProps
}

export const DetailWidget = (props: DetailWidgetProps): React.JSX.Element => {
  const { title, children, asBox } = props
  const designProps = extractDesignProps(props)

  return (
    <Box {...designProps}>
      <Text marginBottom="4px" color="lightGray.600">
        {title}
      </Text>
      {asBox ? <Box>{children}</Box> : <Text>{children}</Text>}
    </Box>
  )
}
