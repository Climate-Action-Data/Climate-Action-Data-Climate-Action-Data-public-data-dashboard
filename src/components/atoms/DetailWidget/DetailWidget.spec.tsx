import { render } from '@testing-library/react'
import { DetailWidget } from './DetailWidget'

const DEFAULT_TEST_TITLE = `My Title`
const DEFAULT_TEST_CONTENT = `My Content`

it(`renders correctly`, () => {
  const { container } = render(<DetailWidget title={DEFAULT_TEST_TITLE}>{DEFAULT_TEST_CONTENT}</DetailWidget>)
  expect(container).toMatchSnapshot()
})
