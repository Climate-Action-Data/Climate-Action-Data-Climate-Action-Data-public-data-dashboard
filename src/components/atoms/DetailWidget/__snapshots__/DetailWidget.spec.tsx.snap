// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="css-0"
  >
    <p
      class="chakra-text css-1xi4xed"
    >
      My Title
    </p>
    <p
      class="chakra-text css-0"
    >
      My Content
    </p>
  </div>
</div>
`;
